### DISCUSSION TITLE
Log panel only shows the last log message when it has multiple messages at the same timestamp

### DISCUSSION THREAD

**Discussion by junhyeokahn** [foxglove/community#753] [Q&A] [answered]
I am publishing multiple Log proto schema messages with a same timestamp.
When I load the saved mcap file to studio’s log panel, it only shows the latest log message.
Is there any way I could see all the messages?

**Comment by defunctzombie** [comment]
Please provide a sample mcap file we can use to reproduce the issue.

**Comment by junhyeokahn** [comment]
Thanks for the reply. 
I was trying to send and visualize multiple logs using foxglove's Log proto, but I just realized that it is only for a single text at a single timestamp by just emulating ROS's log msg.

So, I am now trying to send multiple logs at single timestamp using my custom proto instead of using foxglove's Log proto.
```
syntax = "proto3";

package mrl.logging;

message Text {
  enum Level {
    kUnknwon = 0;
    kDebug   = 1;
    kInfo    = 2;
    kWarn    = 3;
    kError   = 4;
  }

  repeated Data text_data = 1;

  message Data {
    Level level = 1;
    string text = 2;
  }
}
```
I have an `logger_test.mcap` including my custom proto and I am trying to write a studio extension for visualization. I want to visualize similar to the current Log panel, but work with my custom proto. I have almost zero experience on typescript and any help or advice would be highly appreciated!
[logger_test.mcap.zip](https://github.com/foxglove/community/files/11794183/logger_test.mcap.zip)

**Reply by defunctzombie** [reply, marked as answer]
The mcap file you provided has messages of type `mrl.logging.Text`

```
library: libmcap 1.0.0
profile:
messages: 11
duration: 9s
start: 0.000
end: 9.000
compression:
	zstd: [1/1 chunks] [00 MB/00 MB (52.88%)] [0.00 MB/sec]
channels:
  	(1) mrl.logging.Text  11 msgs (1.22 Hz)   : mrl.logging.Text [protobuf]
attachments: 0
metadata: 0
```

Which are not supported by the log panel. However, you said you were seeing only one log message. Since this custom message is not supported by the log panel you would not see any messages. So maybe you did something else to try visualizing your data? Please provide all of the steps you are taking that lead to seeing only one message per timestamp in the log panel.

**Reply by junhyeokahn** [reply]
Hi @defunctzombie , Yes I ended up creating my own custom panel.
Based on this mcap thread, I was able to visualize `currentFrame` message ([This](https://github.com/junhyeokahn/foxglove-extension-practice) is the repo I am working on).

Now, I am trying to extend this to visualize [allFrames](https://github.com/junhyeokahn/foxglove-extension-practice/blob/fdf160d859f1e5bfba4b68561b39f9b601c1d463/src/TextLogPanel.tsx#L79-L85), but couldn't find luck yet. When I try to visualize `allFrames`, it shows an empty message.
Is there something I miss in the code?

**Reply by junhyeokahn** [reply]
This is the visualization result I have in my custom panel. As you can see, the `currentFrame` is printed correctly, but `allFrames` is empty.
![image](https://github.com/foxglove/community/assets/23005104/da208d70-5ee6-4a96-90d6-434e4d70b0d6)

**Reply by junhyeokahn** [reply]
I found I used `useMemo` in the wrong way. Answered in the same slack thread. I will close this!
